spec/system/groups_spec.rb

```diff
@@ -0,0 +1,72 @@
+require_relative '../rails_helper'
+
+RSpec.describe 'Groups', type: :system do
+  let(:user) { User.create(name: 'First User', email: 'test@user.com', password: 'password') }
+  before do
+    # comment the line below to run tests in browser
+    driven_by(:rack_test)
+
+    puts "\nCreating test data..."
+
+    5.times do |i|
+      group = Group.create(name: "Group #{i}", author: user)
+      group.icon.attach(io: File.open(Rails.root.join('spec', 'models', 'files', 'test.jpg')), filename: 'test.jpg',
+                        content_type: 'image/jpeg')
+      group.save
+
+      5.times do |j|
+        entity = Entity.create(name: "Entity #{j}", amount: 100, author: user)
+        entity.save
+      end
+    end
+
+    Group.all.each do |group|
+      group.entities << Entity.all.sample(rand(1..3))
+    end
```

Comment:
I noticed you have some failing tests it would be nice if you fixed this issue. Look at the screenshot below:
![Screenshot 2023-09-11 at 9 20 15 PM](https://github.com/unorjikingsley/Budget-App/assets/79506105/f788062a-9feb-4391-8aa4-36642bd66d00)